metadata-ingestion/src/datahub/ingestion/source/sagemaker.py

```diff
@@ -0,0 +1,294 @@
+
+                full_table_name = f"{glue_database}.{glue_table}"
+
+                self.report.report_warning(
+                    full_table_name,
+                    f"""Note: table {full_table_name} is an AWS Glue object.
+                        To view full table metadata, run Glue ingestion
+                        (see https://datahubproject.io/docs/metadata-ingestion/#aws-glue-glue)""",
+                )
+
+                feature_sources.append(
+                    f"urn:li:dataset:(urn:li:dataPlatform:glue,{full_table_name},{self.source_config.env})"
+                )
+
+        # note that there's also an OnlineStoreConfig field, but this
+        # lacks enough metadata to create a dataset
+        # (only specifies the security config and whether it's enabled at all)
+
+        # append feature name and type
+        if is_record_identifier:
+            primary_key_snapshot: MLPrimaryKeySnapshot = MLPrimaryKeySnapshot(
+                urn=builder.make_ml_primary_key_urn(
+                    feature_group_details["FeatureGroupName"],
+                    feature["FeatureName"],
+                ),
+                aspects=[
+                    MLPrimaryKeyPropertiesClass(
+                        dataType=self.get_feature_type(
+                            feature["FeatureType"], feature["FeatureName"]
+                        ),
+                        sources=feature_sources,
+                    ),
+                ],
+            )
+
+            # make the MCE and workunit
+            mce = MetadataChangeEvent(proposedSnapshot=primary_key_snapshot)
+        else:
+            # create snapshot instance for the feature
+            feature_snapshot: MLFeatureSnapshot = MLFeatureSnapshot(
+                urn=builder.make_ml_feature_urn(
+                    feature_group_details["FeatureGroupName"],
+                    feature["FeatureName"],
+                ),
+                aspects=[
+                    MLFeaturePropertiesClass(
+                        dataType=self.get_feature_type(
+                            feature["FeatureType"], feature["FeatureName"]
+                        ),
+                        sources=feature_sources,
+                    )
+                ],
+            )
+
+            # make the MCE and workunit
+            mce = MetadataChangeEvent(proposedSnapshot=feature_snapshot)
+
+        return MetadataWorkUnit(
+            id=f'{feature_group_details["FeatureGroupName"]}-{feature["FeatureName"]}',
+            mce=mce,
+        )
+
+    def get_workunits(self) -> Iterable[MetadataWorkUnit]:
+
+        feature_groups = self.get_all_feature_groups()
+
+        for feature_group in feature_groups:
+
+            feature_group_details = self.get_feature_group_details(
+                feature_group["FeatureGroupName"]
+            )
+
+            for feature in feature_group_details["FeatureDefinitions"]:
+                yield self.get_feature_wu(feature_group_details, feature)
```

Comment:
make sure to call `self.report.report_workunit()` as well (ditto below)

we might want to make the `report_workunit(wu)` return the workunit, so we can just do `yield self.report.report_workunit(self.get_feature_wu(...))`